docs/architecture/cloud-native/azure-security.md

```diff
@@ -0,0 +1,257 @@
+
+Many security [experts suggest](https://www.troyhunt.com/password-managers-dont-have-to-be-perfect-they-just-have-to-be-better-than-not-having-one/) that using a password manager to keep your own passwords is the best approach. While it centralizes your passwords in one location, it also allows using highly complex passwords and ensuring they're unique for each account. The same system exists within Azure: a central store for secrets.
+
+## Azure Key Vault
+
+Azure Key Vault provides a centralized location to store passwords for things such as databases, API keys, and certificates. Once a secret is entered into the Vault, it's never shown again and the commands to extract and view it are purposefully complicated. The information in the safe is protected using either software encryption or FIPS 140-2 Level 2 validated Hardware Security Modules. 
+
+Access to the key vault is provided through RBACs, meaning that not just any user can access the information in the vault. Say a web application wishes to access the database connection string stored in Azure Key Vault. To gain access, applications need to run using a service principal. Under this assumed role, they can read the secrets from the safe. There are a number of different security settings that can further limit the access that an application has to the vault, so that it can't update secrets but only read them. 
+
+Access to the key vault can be monitored to ensure that only the expected applications are accessing the vault. The logs can be integrated back into Azure Monitor, unlocking the ability to set up alerts when unexpected conditions are encountered. 
+
+## Kubernetes
+
+Within Kubernetes, there's a similar service for maintaining small pieces of secret information. Kubernetes Secrets can be set via the typical `kubectl` executable.
+
+Creating a secret is as simple as finding the base64 version of the values to be stored:
+
+```cli
+echo -n 'admin' | base64
+YWRtaW4=
+echo -n '1f2d1e2e67df' | base64
+MWYyZDFlMmU2N2Rm
+```
+
+Then adding it to a secrets file named `secret.yml` for example that looks similar to the following example:
+
+```yml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mysecret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: MWYyZDFlMmU2N2Rm
+```
+
+Finally, this file can be loaded into Kubernetes by running the following command:
+
+```cli
+kubectl apply -f ./secret.yaml
+```
+
+These secrets can then be mounted into volumes or exposed to container processes through environmental variables. The [12 factor app](https://12factor.net/), covered in [Chapter 1](introduction-to-cloud-native-applications.md), approach to building applications suggests using the lowest common denominator to transmit settings to an application. This is environmental variables as they're supported no matter the operating system or application.
```

Comment:
It seems that the name of the app is Twelve-factor app

Also, it's odd that part of the noun is separated. Is it important to say that was covered in chapter 1? Can that be said separately?

The last sentence also has issues. I imagine environmental variables is supposed to say environment variables but I don't understand how the sentence are connected.

Reply:
I'll clean it up.